src/engines/html5/capabilities/html5-autoplay.js

```diff
@@ -1,35 +1,111 @@
+      Html5AutoPlayCapabilities._autoPlayPromise = Html5AutoPlayCapabilities._vid.play() || Html5AutoPlayCapabilities._oldBrowsersCapability(Html5AutoPlayCapabilities._vid);
+    } catch (e) {
+      Html5AutoPlayCapabilities._autoPlayPromise = Promise.reject();
+    }
+
+  }
+
+  /***
+   * Runs test for muted autoplay capability.
+   * @static
+   * @returns {void}
+   */
+  static _runMutedAutoplayCapability(): void {
     try {
-      Html5AutoPlayCapability._vid.src = EncodingSources.Base64Mp4Source;
-      Html5AutoPlayCapability._promise = Html5AutoPlayCapability._vid.play() || Promise.resolve();
+      Html5AutoPlayCapabilities._mutedVid.src = EncodingSources.Base64Mp4Source;
+      Html5AutoPlayCapabilities._mutedVid.muted = true;
+      Html5AutoPlayCapabilities._mutedAutoPlayPromise = Html5AutoPlayCapabilities._mutedVid.play() || Html5AutoPlayCapabilities._oldBrowsersCapability(Html5AutoPlayCapabilities._mutedVid);
     } catch (e) {
-      Html5AutoPlayCapability._promise = Promise.reject();
+      Html5AutoPlayCapabilities._mutedAutoPlayPromise = Promise.reject();
     }
   }
 
+  /***
+   * Runs the test for autoplay capability in older browsers, where play() does not return a promise
+   * @static
+   * @param {HTMLVideoElement} element - video element to test
+   * @returns {Promise<*>} - return a resolve/reject upon the success of playing
+   */
+  static _oldBrowsersCapability(element: HTMLVideoElement): Promise<*> {
+    return new Promise((resolve, reject) => {
+      let supported = setTimeout(() => {
+        reject();
+      }, playTimeoutBuffer);
+      element.addEventListener(Html5Events.PLAYING, () => {
+        clearTimeout(supported);
+        resolve();
+      });
+      element.addEventListener(Html5Events.ERROR, () => {
+        clearTimeout(supported);
+        reject();
+      });
+    })
+  }
+
+  /***
+   * act as a proxy to the promises. Promise.all is rejected if one of the promises is rejected, and the other promises
+   * data is lost, so here, we always resolve the promise, with status false.
+   * @param {Promise<*>} promise - one of the capabilities result.
+   * @returns {Promise<*>} - promise after manipulation.
+   * @static
+   * @public
+   */
+  static _reflect(promise: Promise<*>): Promise<*> {
+    return promise
+      .then(data => ({data: data, status: true}))
+      .catch(error => ({error: error, status: false}))
+  }
+
   /**
-   * Gets the test result for autoplay capability.
+   * Gets test result for autoplay & muted autoplay capabilities.
    * @returns {Promise<CapabilityResult>} - The result object for autoplay capability.
    * @static
    * @public
    */
   static getCapability(): Promise<CapabilityResult> {
-    return Html5AutoPlayCapability._promise
-      .then(() => ({autoplay: true}))
-      .catch(() => ({autoplay: false}));
+    const promiseArr = [Html5AutoPlayCapabilities._autoPlayPromise, Html5AutoPlayCapabilities._mutedAutoPlayPromise];
```

Comment:
I think this better be cascading check, e.g. first try autoplay and only then muted autoplay cause if autoplay is supported we don't need to try muted autoplay

Reply:
K, will change it.